lib/browser_client.dart

```diff
@@ -73,6 +73,12 @@ class BrowserClient extends BaseClient {
           StackTrace.current);
     });
 
+    xhr.onAbort.first.then((_) {
```

Comment:
I'm guessing we throw this error so that in-flight request futures are guaranteed to complete? If so, add a comment indicating that.

What's the behavior of the IO client if you close it while a request is in-flight?

Reply:
As an example.

```
import 'dart:async';
import 'package:http/http.dart';

main() async {
  var client = new Client();
  var future = client.get('https://httpbin.org/delay/1000');

  print('making request');
  future.then((response) {
    print(response.statusCode);
  }).catchError((error) {
    print(error);
  });

  new Timer(new Duration(seconds: 1), () {
    print('closing');
    client.close();
  });
}
```
Output:
```shell
making request
closing
Connection closed before full header was received
```